source/base/patterns.cc

```diff
@@ -1047,6 +1047,8 @@ namespace Patterns
             return description.str();
           }
       }
+      break;
```

Comment:
This may fix the warning, but it's the wrong fix -- the `return` is misplaced in line 1047 and needs to be moved after the loop. At least I'm pretty sure that's the case -- @luca-heltai ?

Reply:
agreed.

Reply:
Thanks for pointing. Now, I address the real problem in this PR.